go/adbc/driver/databricks/record_builder.go

```diff
@@ -0,0 +1,338 @@
+package databricks
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/apache/arrow-go/v18/arrow"
+	"github.com/apache/arrow-go/v18/arrow/array"
+	"github.com/apache/arrow-go/v18/arrow/memory"
+)
+
+// ColumnBuilder is an interface for building Arrow arrays
+type ColumnBuilder interface {
+	Append(value interface{}) error
+	Builder() array.Builder
+}
+
+// StringColumnBuilder handles string values
+type StringColumnBuilder struct {
+	builder *array.StringBuilder
+}
+
+func NewStringColumnBuilder(mem memory.Allocator) *StringColumnBuilder {
+	return &StringColumnBuilder{
+		builder: array.NewStringBuilder(mem),
+	}
+}
+
+func (b *StringColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	if str, ok := value.(string); ok {
+		b.builder.Append(str)
+	} else {
+		b.builder.Append(fmt.Sprintf("%v", value))
```

Comment:
Not a good idea. If the value doesn't cast to string, something is wrong with our code and this fallback is not ok.